src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +881,71 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+
+int AutoDJCratesDAO::getRandomTrackIdFromLibrary(void) {
+
+    // getRandomTrackId() would have already created
+    // the temporary auto-DJ-crates database.
+    QSqlQuery oQuery(m_rDatabase);
+
+    int iTrackId = -1;
+    // Get total as well as unplayed tracks
+    int iUnplayedTracks = 0, iTotalTracks = 0;
+    oQuery.prepare("SELECT COUNT(*) AS count FROM library "
+          "WHERE timesplayed = 0 "
+          "UNION ALL SELECT COUNT(*) AS count FROM "
+          "library ");
+    if (oQuery.exec()) {
+        if (oQuery.next()) {
+            iUnplayedTracks = oQuery.value(0).toInt();
+            if (oQuery.next())
+                iTotalTracks = oQuery.value(0).toInt();
+        }
+    } else {
+        LOG_FAILED_QUERY(oQuery);
+        return -1;
+    }
+
+    qDebug() << "total "<< iTotalTracks<< "unplayed " << iUnplayedTracks;
+
+    if ( iUnplayedTracks != 0) {
+        // Get an unplayed track id
+        oQuery.prepare("SELECT library.id  "
+                "FROM library "
+                "WHERE timesplayed = 0 "
+                "ORDER BY RANDOM() "
```

Comment:
Have not checked it, but will this work? 
```
oQuery.prepare("SELECT library.id "
               "FROM library "
               "ORDER BY timesplayed , random() "
               "LIMIT 1")
```

Just found this, It here probably an issue? : 
http://stackoverflow.com/questions/1253561/sqlite-order-by-rand